File: tests/nodejs/cases/case_124_indirect_container_write_03.js
/**
 * Task case 124: indirect_container_write_03 false-positive/false-negative stress pattern.
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case124IndirectContainerWrite03(input) {
  const taskName = 'indirect_container_write_03';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'stress',
    stage: 'evaluation',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  const envelope = { wrapped: payload };
  // ESCAPE: payload stored via indirection into retained sink.
  retainedAudit.push(envelope.wrapped);
  return 'ok';
}

module.exports = {
  case124IndirectContainerWrite03,
};

File: tests/go/cases/case_124_indirect_container_write_03.go
package escape_tests

var retainedCase124 = []map[string]string{}

func Case124IndirectContainerWrite03(input string) string {
	raw := input
	if raw == "" {
		raw = "sample"
	}
	payload := map[string]string{
		"task": "indirect_container_write_03",
		"entity": "stress",
		"stage": "evaluation",
		"input": raw,
		"checksum": "indirect_container_write_03:" + raw,
	}
	envelope := map[string]map[string]string{"wrapped": payload}
	// ESCAPE: indirection writes payload into retained sink.
	retainedCase124 = append(retainedCase124, envelope["wrapped"])
	return "ok"
}
